consensus/src/runway/mod.rs

```diff
@@ -295,13 +278,27 @@ where
         self.keychain.index()
     }
 
-    fn on_unit_received(&mut self, unit: UncheckedSignedUnit<H, D, MK::Signature>) {
-        match self.validator.validate(unit, &self.store) {
-            Ok(unit) => self.send_unit_for_reconstruction(unit),
-            Err(e) => self.handle_validation_error(e),
+    fn handle_dag_result(&mut self, result: DagResult<H, D, MK>) {
+        let (units, requests, alerts) = result.into();
```

Comment:
This conversion I like more, but I also think it doesn't add much clarity. Seeing this line without seeing the implementation of `From` is not immediately clear bc hidden behind `into()` may be some logic. I don't see this conversion having any positive impact on clarity, other than saving ~3 lines of code. Could've just deconstruct the struct (likely need to make its fields `pub`).

Again, just my personal preference. You can choose to resolve.

Reply:
Oh, yeah, I really don't like making fields `pub`, it muddles the interfaces.

Reply:
Idk, to me they seem aight (in case of `*Result` structs here).